libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -606,6 +606,88 @@ export class TeamsActivityHandler extends ActivityHandler {
         }
     }
 
+    /**
+     * Override this method to change the dispatching of MessageUpdate activities.
+     *
+     * @param context A context object for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async dispatchMessageUpdateActivity(context: TurnContext): Promise<void> {
+        if (context.activity.channelId == 'msteams') {
+            const channelData = context.activity.channelData as TeamsChannelData;
+
+            switch (channelData.eventType) {
+                case 'undeleteMessage':
+                    return await this.onTeamsMessageUndelete(context);
+
+                case 'editMessage':
+                    return await this.onTeamsMessageEdit(context);
+
+                default:
+                    return super.dispatchMessageUpdateActivity(context);
+            }
+        } else {
+            return await super.dispatchMessageUpdateActivity(context);
+        }
+    }
+
+    /**
+     * Override this method to change the dispatching of MessageDelete activities.
+     *
+     * @param context A context object for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async dispatchMessageDeleteActivity(context: TurnContext): Promise<void> {
```

Comment:
@singhk97 I appreciate the changes to ActivityHandler and ActivityHandlerBase for ActivityTypes `messageUpdate` and `messageDelete`, but I'm wondering if these changes should be localized in TeamsActivityHandler.

@boydc2014 two questions, does Direct Line support sending these two ActivityType and do other, non-premium channels support these ActivityTypes as well? e.g., Slack

@compulim does Web Chat innately support messageUpdate and messageDelete initialized by the end-user or is this an available extensiblity point?

@johnataylor FYI

Reply:
Since we can't know if these are generic or not, we will roll with generic handling in ActivityHandlerBase.  If they are in fact not generic, that code won't be reached.  If they are only Teams, the TeamsActivityHandler will handle it.